dao/src/main/java/org/wso2/testgrid/dao/EntityManagerHelper.java

```diff
@@ -89,10 +95,34 @@ public static void closeEntityManager(String persistenceUnitName) {
      * @return entity manager factory with the given persistence unit name
      */
     private static EntityManagerFactory getEntityManagerFactory(String persistenceUnitName) {
+
+        Map<String, String> persistenceMap = new HashMap<String, String>();
         EntityManagerFactory entityManagerFactory = entityManagerFactoryMap.get(persistenceUnitName);
         if (entityManagerFactory == null) {
-            entityManagerFactory = Persistence.createEntityManagerFactory(persistenceUnitName);
-            entityManagerFactoryMap.put(persistenceUnitName, entityManagerFactory);
+            try {
+                //Get database info from testgrid-config.properties
+                String dbUrl = ConfigurationContext.getProperty(Constants.DB_URL);
+                String dbUser = ConfigurationContext.getProperty(Constants.DB_USER);
+                String dbUserPass = ConfigurationContext.getProperty(Constants.DB_USER_PASS);
+
+                persistenceMap.put("javax.persistence.jdbc.url", dbUrl);
+                persistenceMap.put("javax.persistence.jdbc.user", dbUser);
+                persistenceMap.put("javax.persistence.jdbc.password", dbUserPass);
+                if (dbUrl != null && dbUser != null && dbUserPass != null) {
+                    //Override properties taken from persistence.xml
+                    entityManagerFactory = Persistence.createEntityManagerFactory(persistenceUnitName, persistenceMap);
+                } else {
+                    logger.warn("One or more database properties in testgrid-config.properties are null. " +
+                            "Using default properties in persistence.xml");
+                }
+            } catch (TestGridException e) {
```

Comment:
I think throwing exceptions from the ConfigurationContext#getProperty was not correct. Shall we dig in and improve that part of the code? I think then this code logic will be much cleaner.